consensus/istanbul/backend/announce.go

```diff
@@ -937,7 +937,7 @@ func (vc *versionCertificate) payloadToSign() ([]byte, error) {
 func (sb *Backend) generateVersionCertificate(version uint) (*versionCertificate, error) {
 	vc := &versionCertificate{
 		Address:   sb.Address(),
-		PublicKey: sb.publicKey,
+		PublicKey: sb.authorizeInfo.Load().(*AuthorizeInfo).PublicKey,
 		Version:   version,
 	}
 	err := vc.Sign(sb.Sign)
```

Comment:
There are 3 race conditions here, we do an atomic load for the Address the public key and for the sign operation. Between each load Authorize could have been called. We can solve this by just loading at the top of the function and making sure we reference all values from the loaded instance.

Reply:
You are right! Nice catch